lib/voxpupuli/acceptance/spec_helper_acceptance.rb

```diff
@@ -1,4 +1,31 @@
-def configure_beaker(modules: :metadata, &block)
+ENV_VAR_PREFIX = 'BEAKER_FACTER_'
+FACT_FILE = '/etc/facter/facts.d/voxpupuli-acceptance-env.json'
+
+def beaker_facts_from_env
+  facts = {}
+
+  ENV.each do |var, value|
+    next unless var.start_with?(ENV_VAR_PREFIX)
+
+    fact = var.sub(ENV_VAR_PREFIX, '').downcase
+    facts[fact] = value
+  end
+
+  facts
+end
+
+def write_beaker_facts_on(hosts)
+  beaker_facts = beaker_facts_from_env
+
+  if beaker_facts.any?
+    require 'json'
+    on(hosts, "mkdir -p #{File.dirname(FACT_FILE)} && cat <<VOXPUPULI_BEAKER_ENV_VARS > #{FACT_FILE}\n#{beaker_facts.to_json}\nVOXPUPULI_BEAKER_ENV_VARS")
```

Comment:
took me a little while to wrap my head round this bash heredoc, but it does look correct.

Reply:
Yes, it's a bit obscure. However, the alternative was using `scp_to` and that wasn't great either. Ideally beaker would have something like `write_to_file(destination, content)`.